libindex/libindex.go

```diff
@@ -158,3 +161,58 @@ func (l *Libindex) IndexReport(ctx context.Context, hash claircore.Digest) (*cla
 	res, ok, err := l.store.IndexReport(ctx, hash)
 	return res, ok, err
 }
+
+// AffectedManifests retrieves a list of affected manifests when provided a list of vulnerabilities.
+func (l *Libindex) AffectedManifests(ctx context.Context, vulns []claircore.Vulnerability) ([]claircore.Digest, error) {
+	const (
+		maxGroupSize = 100
+	)
+	log := zerolog.Ctx(ctx).With().
+		Str("component", "libindex/Libindex.AffectedManifests").
+		Logger()
+
+	var manifests struct {
+		sync.Mutex
+		m []claircore.Digest
+	}
+
+	groupSize := int(math.Sqrt(float64(len(vulns))))
+	if groupSize > maxGroupSize {
+		groupSize = maxGroupSize
+	}
```

Comment:
Why do this instead of a fixed unit of work?

Reply:
my thinking is creating equal sized concurrency groups proportionate to the length of the slice coming in allows a balance between concurrency and database connection usage (to some max).